crates/orderbook/src/order_quoting.rs

```diff
@@ -223,6 +225,9 @@ pub struct QuoteData {
     pub fee_parameters: FeeParameters,
     pub kind: OrderKind,
     pub expiration: DateTime<Utc>,
+    /// Since different quote kinds have different expirations,
+    /// we need to store the quote kind to prevent missuse of quotes.
+    pub quote_kind: QuoteKind,
```

Comment:
ubernit: Same as other comment
```suggestion
    pub kind: QuoteKind,
```

Reply:
3 lines above we have already declared a kind variable `kind` for Orderkind. Yes, it feels strange, but I think we need to keep it.